packages/payload/src/utilities/createLocalReq.ts

```diff
@@ -56,7 +57,11 @@ export const createLocalReq: CreateLocalReq = (
   req.i18n = i18n
   req.t = i18n.t
   req.user = user || req?.user || null
-  if (!req.collection) req.collection = collection ? payload.collections?.[collection] : null
+  req.collection = collection
+    ? payload.collections?.[collection]
+      ? isolateObjectProperty(payload.collections[collection], collection)
```

Comment:
If I understand this right, `isolateObjectProperty` might not be having the desired outcome here. `collection` is argument which contains the collection name. that property won't exist inside the `payload.collections[collection]` object.

Also by overwriting `req.collection` we get back to the original issue.